test/functional/wallet_sweep.py

```diff
@@ -0,0 +1,172 @@
+        self.generate_initial_utxos([8, 13])
+        wallet_balance_before_sweep = self.wallet.getbalances()["mine"]["trusted"]
+
+        sweep_tx_receipt = self.wallet.sweep([{self.return_addr_with_amount: 5}, self.return_addr_remainder])
+        self.generate(self.nodes[0], 1)
+        assert_equal(0, self.wallet.getbalances()["mine"]["trusted"]) # wallet is empty
+
+        assert_equal(sweep_tx_receipt["complete"], True)
+        tx_from_wallet = self.wallet.gettransaction(txid = sweep_tx_receipt["txid"], verbose = True)
+
+        assert_equal(len(tx_from_wallet["decoded"]["vout"]), 2)
+        self.assert_tx_has_output(tx_from_wallet, self.return_addr_with_amount, 5)
+        self.assert_tx_has_output(tx_from_wallet, self.return_addr_remainder)
+
+        self.assert_balance_swept_completely(tx_from_wallet, wallet_balance_before_sweep)
+
+    def sweep_invalid_receiver_addresses(self):
+        self.log.info("Testing sweep only with specified amount")
+        self.generate_initial_utxos([12, 9])
+        assert_greater_than(self.wallet.getbalances()["mine"]["trusted"], 0)
+
+        assert_raises_rpc_error(-8, "Must provide at least one address without a specified amount" , self.wallet.sweep, [{self.return_addr_with_amount: 5}])
+
+        self.clean_up()
+
+    def sweep_invalid_amounts(self):
+        self.log.info("Try sweeping more than balance")
+        self.generate_initial_utxos([7, 14])
+        wallet_balance_before_sweep = self.wallet.getbalances()["mine"]["trusted"]
+
+        expected_tx = self.wallet.sweep(receivers=[{self.return_addr_with_amount: 5}, self.return_addr_remainder], options={"add_to_wallet": False})
+        tx = self.wallet.decoderawtransaction(expected_tx['hex'])
+        fee = 21 - sum([o["value"] for o in tx["vout"]])
+
+        assert_raises_rpc_error(-8, "Assigned more value to outputs than available funds." , self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep + 1}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Insufficient funds for fees after creating specified outputs.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep}, self.return_addr_remainder])
+        assert_raises_rpc_error(-8, "Specified output amount to {} is below dust threshold".format(self.return_addr_with_amount), self.wallet.sweep, [{self.return_addr_with_amount: 0.00000001}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Dynamically assigned remainder results in dust output.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep - fee}, self.return_addr_remainder])
+        assert_raises_rpc_error(-6, "Dynamically assigned remainder results in dust output.", self.wallet.sweep, [{self.return_addr_with_amount: wallet_balance_before_sweep - fee - Decimal(0.00000010)}, self.return_addr_remainder])
+
+        self.clean_up()
+
+    def sweep_negative_effective_value(self):
```

Comment:
I think you can add a `self.log.info(..)` statement here.

Reply:
Done